flink-end-to-end-tests/flink-bucketing-sink-test/src/main/java/org/apache/flink/streaming/tests/BucketingSinkTestProgram.java

```diff
@@ -143,6 +147,7 @@ public void open(Configuration parameters) {
 		private final int durationMs;
 
 		private long ms = 0;
+		private volatile boolean canceled = false;
```

Comment:
Is the volatile really necessary? I thought every task is single threaded?

Reply:
There are actually asynchronous calls that are executed from different threads.